File: app/src/main/java/com/rayworks/droidweekly/viewmodel/ArticleListBindings.java
package com.rayworks.droidweekly.viewmodel;

import android.databinding.BindingAdapter;
import android.support.v7.widget.RecyclerView;

import com.rayworks.droidweekly.ArticleAdapter;
import com.rayworks.droidweekly.model.ArticleItem;

import java.util.List;

public class ArticleListBindings {
    @SuppressWarnings("unchecked")
    @BindingAdapter("app:items")
    public static void setItems(RecyclerView listView, List<ArticleItem> items) {
        ArticleAdapter adapter = (ArticleAdapter) listView.getAdapter();

        if (adapter != null) {
            adapter.update(items);
        }
    }
}

File: app/src/main/java/com/rayworks/droidweekly/viewmodel/ArticleListBindings.kt
package com.rayworks.droidweekly.viewmodel

import androidx.databinding.BindingAdapter
import androidx.recyclerview.widget.RecyclerView
import com.rayworks.droidweekly.model.ArticleItem
import com.rayworks.droidweekly.ui.component.ArticleAdapter

/***
 * The custom binding for updating the data of a [RecyclerView.Adapter].
 */
object ArticleListBindings {
    /***
     * Sets the binding list data
     */
    @JvmStatic
    @BindingAdapter("app:items")
    fun setItems(
        listView: RecyclerView,
        items: List<ArticleItem>?,
    ) {
        val adapter = listView.adapter as ArticleAdapter?
        adapter?.update(items)
    }
}
